### PR TITLE
Cleanup ReadAsync calls, make IISHttpContext private

### PR DESCRIPTION
**Pull request by jkotalik**
Accidentally made a PR to dev. I reverted it and am now doing the PR to release/2.1

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs [hunk 1]
```diff
@@ -39,7 +39,9 @@ public override int Read(byte[] buffer, int offset, int count)
 
         public override unsafe Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            return _httpContext.ReadAsync(buffer, offset, count, cancellationToken);
+            var memory = new Memory<byte>(buffer, offset, count);
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: So what does happen when buffer is null or count is zero?

Comment (jkotalik) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: We allow it. After chatting with @davidfowl, a use case could be that someone calls ReadAsync without a buffer to know when data is available. This reduces the amount of idle allocated memory.

Comment (Tratcher) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: Does it actually work? You removed the tests.

Comment (jkotalik) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: Yeah, I'll need to verify this doesn't null ref.

Comment (davidfowl) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: Why would it null ref?

**Comment by jkotalik** [comment]
🆙 📅

**Comment by jkotalik** [comment]
@pakrym can you take a look at this? This is some general reaction that Fowler didn't like about https://github.com/aspnet/IISIntegration/commit/5e74b36c53038116c6a789a1e2a9b084f59dc377

File: src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs (hunk 1 above)
Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: Memory supports null buffer, it'll return empty span.

**Comment by jkotalik** [comment]
I reverted it in dev to first merge into release. Will merge into dev afterwards.

**Pull request merged**